dev/tools/localization/bin/gen_date_localizations.dart

```diff
@@ -139,17 +139,17 @@ String _jsonToMap(dynamic json) {
   if (json == null || json is num || json is bool)
     return '$json';
 
-  if (json is String) {
-    if (currentLocale == 'kn')
-      return generateEncodedString(json);
-    else if (json.contains("'"))
-      return 'r"""$json"""';
-    else
-      return "r'''$json'''";
-  }
+  if (json is String)
+    return generateEncodedString(currentLocale, json);
```

Comment:
@a14n A separate question, but this function was originally created for use to handle the Kannada locale. What's the rationale for changing its purpose here? It also seems like the documentation for the function have not been appropriately updated so I'm not quite sure what the intent is

Reply:
`generateEncodedString` now manages both Kannada and not-Kannada. The test on locale is done at a single place.

>It also seems like the documentation for the function have not been appropriately updated

Indeed it should now mention that the method handles both Kannada and not-Kannada. Perhaps a better function name could be found.